pkg/routeagent/controllers/route/route.go

```diff
@@ -826,3 +916,65 @@ func (r *Controller) reconcileRoutes(vxlanGw net.IP) error {
 	}
 	return nil
 }
+
+func (r *Controller) configureIPRule(operation Operation) error {
+	if r.cniIface != nil {
+		rule := netlink.NewRule()
+		rule.Table = RouteAgentHostNetworkTableID
+		rule.Priority = RouteAgentHostNetworkTableID
+
+		switch operation {
+		case AddRoute:
+			err := netlink.RuleAdd(rule)
+			if err != nil && !os.IsExist(err) {
+				return fmt.Errorf("failed to add ip rule %s: %v", rule.String(), err)
+			}
+		case DeleteRoute:
+			err := netlink.RuleDel(rule)
+			if err != nil && !os.IsNotExist(err) {
+				return fmt.Errorf("failed to delete ip rule %s: %v", rule.String(), err)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *Controller) configureRoute(remoteSubnet string, operation Operation) error {
+	src := net.ParseIP(r.cniIface.ipAddress)
+	_, dst, err := net.ParseCIDR(remoteSubnet)
+	if err != nil {
+		return fmt.Errorf("error parsing cidr block %s: %v", remoteSubnet, err)
+	}
+
+	ifaceIndex := r.defaultHostIface.Index
+	// TODO: Add support for this in the CableDrivers themselves.
```

Comment:
Perhaps we could have the cable driver optionally publish its interface index via the Endpoint's BackendConfig.

Reply:
It's going to be more complicated, we will need implication from the cable driver in some cases, for example. Some cable drivers (ip on ip, ... etc) could need multiple interfaces, one per remote cluster. That would mean that we need to decide about the interface based on the remote cluster.